### DISCUSSION TITLE
🚀 FR RTE/cKeditor better support of linking, especially to Structure urls

### DISCUSSION THREAD

**Discussion by robinsowell** [ExpressionEngine/ExpressionEngine#3888] [Feature Requests]
### Suggest an idea for ExpressionEngine

Came up in slack, discussing what would keep folks on wygwam instead of native rte/ckeditor in EE7.  Main thing looks to be link handling:

> 
> The other thing Wygwam has a big functional leg up on is linking. Especially when you use Structure. With Wygwam you click to add a link, a modal comes up and it allows you to have a dropdown where you can select from your static pages, any listing channels, a pasted in URL or using the file browser to link to a doc. Then you have the various link controls (like opening in a new window). If you have selected a Structure static page or channel entry it actually stores the link ID so if you change the URL later for that linked page, it will automatically update. Pretty nice!
> With the CKEditor in RTE you only get a field to paste in a URL or you can use the somewhat hidden trick of using the @ and start typing. But that assumes you know the name of the page. Overall a pretty clunky experience for editors compared with what is in Wygwam.
> That's a big barrier, at least for me, to going native RTE. I really want to go native.

**Comment by intoeetive** [comment]
From internal discussion in Slack:
There's no "almost ready" solution that we could extend, but I found this thread on GitHub which has some good examples
https://github.com/ckeditor/ckeditor5/issues/4836

**Reply by intoeetive** [reply]
There's some work-in-progress in `feature/7.x/ckeditor-link-to-pages` which did not get to the point of working concept yet... posting it here so it wouldn't get lost

**Comment by ataucher** [comment]
This was a main reason for me to stay on wygwam, but now with the EE 7.4.11 update, the "link type" in the wygwam modal reverts back to a static URL instead of a "structure page". So when migrating a local dev database to staging or production, all the links created with wygwam using structure pages, are all now reverted to static dev urls – link id is not retained. 
Coincides with this issue #[4387](https://github.com/ExpressionEngine/ExpressionEngine/issues/4387)
